code/game/objects/structures/gym.dm

```diff
@@ -0,0 +1,216 @@
+/obj/structure/punching_bag
+	name = "punching bag"
+	desc = "A punching bag. Can you get to speed level 4???"
+	icon = 'icons/obj/structures/fitness.dmi'
+	icon_state = "punchingbag"
+	anchored = TRUE
+	layer = WALL_OBJ_LAYER
+	var/list/hit_sounds = list(
+		'sound/weapons/genhit1.ogg',
+		'sound/weapons/genhit2.ogg',
+		'sound/weapons/genhit3.ogg',
+		'sound/weapons/punch1.ogg',
+		'sound/weapons/punch2.ogg',
+		'sound/weapons/punch3.ogg',
+		'sound/weapons/punch4.ogg'
+	)
+
+/obj/structure/punching_bag/attack_hand(mob/user)
+	if(!ishuman(user))
+		return
+	flick("[icon_state]2", src)
+	playsound(loc, pick(hit_sounds), 25, TRUE, 3)
+
+/obj/structure/punching_bag/attack_alien(mob/living/carbon/xenomorph/M)
+	M.animation_attack_on(src)
+	playsound(src, 'sound/effects/metalhit.ogg', 25, 1)
+	M.visible_message(SPAN_DANGER("[M] slices \the [src] apart!"),
```

Comment:
fix up the single letter vars 
https://github.com/cmss13-devs/cmss13/blob/master/.github/guides/STYLES.md#dont-use-abbreviations